sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -59,31 +65,120 @@ public HttpPipeline getHttpPipeline() {
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, payload)
+            .flatMap(
+                response -> {
+                    try {
+                        return Mono.just(mapper.writeValueAsString(response.getValue()));
+                    } catch (JsonProcessingException e) {
+                        return Mono.error(e);
+                    }
+                });
+    }
+
+    // Input and output are Object.
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Object> createDigitalTwinObject(String digitalTwinId, Object digitalTwin) {
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, digitalTwin)
+            .flatMap(
+                response -> Mono.just(response.getValue()));
+    }
+
+    // Input and output are T (Generics).
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<T> createDigitalTwinGeneric(String digitalTwinId, Object digitalTwin, Class<T> klazz) {
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, digitalTwin)
+            .flatMap(
+                response -> Mono.just(mapper.convertValue(response.getValue(), klazz)));
+    }
+
+    // Input is String and output is Response<String>.
+    // TODO: Autorest team -> the etag returned by the service is present under both Response.getHeaders() and ResponseBase.deserializedHeaders().
+    // TODO: (cont.) Since etag is a well known http header, it should be available via Response.getHeaders(), which it is.
+    // TODO: (cont.) So there shouldn't be a need to define DigitalTwinsAddHeaders explicitly again, and map it to ResponseBase.deserializedHeaders.
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<String>> createDigitalTwinWithResponseString(String digitalTwinId, String digitalTwin) throws JsonProcessingException {
+        Object payload = mapper.readValue(digitalTwin, Object.class);
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, payload)
+            .flatMap(
+                response -> {
+                    try {
+                        String jsonResponse = mapper.writeValueAsString(response.getValue());
+                        return Mono.just(new SimpleResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse));
+                    } catch (JsonProcessingException e) {
+                        return Mono.error(e);
+                    }
+                });
+    }
+
+    // Input is String and output is Response<String> -> ResponseBase<DigitalTwinsAddHeaders, String>.
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<ResponseBase<DigitalTwinsAddHeaders, String>> createDigitalTwinWithResponseResponseBaseString(String digitalTwinId, String digitalTwin) throws JsonProcessingException {
```

Comment:
Option 2:
Accept request as string, deserialize it to object and send to PL.
PL returns response as an object, serialize it to string and return to user.

Response type returned is `ResponseBase<H, T>` - it contains the etag info in explicitly defined `deserializedHeaders`.

Reply:
The only difference in retrieving the etags:

`Response<T>` - response.getHeaders().get("etag")
`Response<H, T>` - response.getDeserializedHeaders().getETag()

Reply:
This is a language call taken by java - not all services return all headers, and they did not want to return `null` for headers where the value is not set; so the raw headers are accessible as a property bag, and no specific getters are implemented in `Response<T>.headers`.

Reply:
so basically the return type in this case is `DigitalTwinsAddRelationshipResponse` which translates into `ResponseBase<DigitalTwinsAddHeaders, String>` is that correct?
So technically as Srikanta mentioned, we can just keep using the simplified version (`DigitalTwinsAddRelationshipResponse` ) is that right?

Reply:
which is basically option 3 in your PR

Reply:
Yup, correct.
We need to decide if we prefer:
`Response<String>` -> response.getHeaders().get("etag")
vs
`DigitalTwinsAddResponse` -> response.getDeserializedHeaders().getETag()